packages/expo-sqlite/src/Storage.ts

```diff
@@ -183,6 +208,28 @@ export class SQLiteStorage {
     }
   }
 
+  /**
+   * Retrieves the number of key-value pairs stored in the storage synchronously.
+   */
+  getLengthSync(): number {
+    const db = this.getDbSync();
+    const result = db.getFirstSync<{ count: number }>(STATEMENT_LENGTH);
+    return result?.count ?? 0;
```

Comment:
We should be always guaranteed exactly 1 row:
```suggestion
    return result!.count ?? 0;
```